### PR TITLE
Bitmex: Fix handling index records in WS trade stream

### LINKED ISSUE #1684: Bitmex: Bug handling index records in WS trade stream
Comment and code conflict at bitmex_websocket.go L218 in `wsHandleData`:
```
			if tradeHolder.Data[i].Price == 0 {
				// Please note that indices (symbols starting with .) post trades at intervals to the trade feed.
				// These have a size of 0 and are used only to indicate a changing price.
				continue
```
_Originally posted by @shazbert in https://github.com/thrasher-corp/gocryptotrader/pull/1586#discussion_r1803912630_

### PR DESCRIPTION
**Pull request by gbjk**
Fixes handling for Size == 0 index records sent to trade stream

Fixes #1684 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run
- [x] TestWsTrades

### CODE REVIEW

**Review by gloriousCode — APPROVED** [review]
I do appreciate the fix. Its just that we don't actually have support in place because [WS support was removed for index](https://github.com/thrasher-corp/gocryptotrader/blame/2232340d499b108a93f02d4447167bebce976542/exchanges/bitmex/bitmex_wrapper.go#L73)
In addition, I needed to add support for the "." prefix for index subscriptions.

After those I can confirm the fix works with real data
image

**Review by shazbert — CHANGES_REQUESTED** [review]
One thing from me and when you have time please rebase. Thanks!

File: exchanges/exchange.go [hunk 1]
```diff
@@ -49,13 +49,15 @@ const (
 	DefaultWebsocketOrderbookBufferLimit = 5
 )
 
+// Public Errors
 var (
-	// ErrExchangeNameIsEmpty is returned when the exchange name is empty
-	ErrExchangeNameIsEmpty = errors.New("exchange name is empty")
+	ErrExchangeNameIsEmpty   = errors.New("exchange name is empty")
+	ErrSymbolCannotBeMatched = errors.New("symbol cannot be matchedX")
```
[the remark this answers is not in the document]
Comment (gbjk) [reply exchanges/exchange.go]: Fixed [9ac5b93](https://github.com/thrasher-corp/gocryptotrader/pull/1685/commits/9ac5b9398cc48d852609b1c77499c008bdb6e359)

**Comment by gbjk** [comment]
> One thing from me and when you have time please rebase. Thanks!

Had the stack ready for this. All done :)

**Review by thrasher- — APPROVED** [review]
Thanks for looking into this

**Pull request merged**